src/general/MainMenu.cs

```diff
@@ -752,6 +788,47 @@ private void WarnAboutNoEnabledMods()
         }
     }
 
+    private float TrackMenuPerformance()
+    {
+        var currentFrameRate = Engine.GetFramesPerSecond();
+
+        // If this is the first tracked frame, do not use the average of the frame delta and 0
+        if (averageFrameRate == 0)
+            return currentFrameRate;
+
+        // Not an exact average by any means, but good enough for this purpose
+        return (averageFrameRate + currentFrameRate) / 2;
+    }
+
+    private void WarnAboutLowPerformance()
+    {
+        if (averageFrameRate < 30 && secondsInMenu >= 16 && !AreAnyMenuPopupsOpen())
```

Comment:
Could you add one more check here to ensure the options menu is not visible? Because saving the options due to the low performance swap when the options menu is open will probably cause issues...